server/game/cards/01_SOR/events/Restock.ts

```diff
@@ -0,0 +1,28 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { TargetMode, WildcardRelativePlayer, ZoneName } from '../../../core/Constants';
+
+export default class Restock extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '6087834273',
+            internalName: 'restock'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Choose up to 4 cards in a discard pile. Put them on the bottom of their owner\'s deck in a random order',
+            targetResolver: {
+                mode: TargetMode.UpTo,
+                numCards: 4,
+                zoneFilter: ZoneName.Discard,
+                controller: WildcardRelativePlayer.Any,
```

Comment:
I think we need to add `canChooseNoCards: true` like in Medal Ceremony. To confirm, let's add a test case for selecting 0 cards